<div id="instructions">
<div>
    <h1>
        <a name="_Contents"></a>1 Contents
    </h1>
</div>
<h4 style="padding-left: 1em;">
    <a href="#_Instructions">2 Instructions</a>
</h4>
<p style="padding-left: 3em;">
    <a href="#_Selecting_a_dataset">2.1 Selecting a dataset</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Loading_a_peak_file">2.2 Loading a peak file</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Looking_at_the_Correlation_table">2.3 Looking at the Correlation table</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Static_heatmap">2.4 Static heatmap</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Responsive_heatmap">2.5 Responsive heatmap</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Tree">2.6 Tree</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Samples_metadata">2.7 Sample metadata</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Correlation_correction">2.8 Correlation correction</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_About_the_datasets">2.9 About the datasets</a>
</p>
<h4 style="padding-left: 1em;">
    <a href="#_FAQ">3 FAQ</a>
</h4>
<p style="padding-left: 3em;">
    <a href="#_How_to_cite">3.1 How to cite?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Where_can_I_find_the_source_code">3.2 Where can I find the source code?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Can_you_add_this_dataset_on_HeatChIPseq">3.3 Can you add this dataset on HeatChIPseq? One of your datasets does not seem up to date, can you update it?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Could_you_implement_this_new_feature">3.4 Could you implement this new feature? | I want to report a bug.</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_I_have_uploaded_a_peak_file">3.5 I have uploaded a peak file. How do I remove it?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_What_happens_to_the_files_I_upload">3.6 What happens to the files I upload?</a>
</p>
<h4 style="padding-left: 1em;">
    <a href="#_About">4 About</a>
</h4>
<img src="screenshot.png" alt="HeatChIPseq screenshot" width="800"> 
<div>
    <h1>
        <a name="_Instructions"></a>2 Instructions
    </h1>
</div>
<p>
    To access the application, click on the <strong>Use Application</strong> button on the <strong>top bar</strong>.
	The step-by-step instructions for using the application are below.
</p>
<h3>
    <a name="_Selecting_a_dataset"></a>2.1 Selecting a dataset
</h3>
<p>
    The first step is to <strong>select a dataset</strong> to work with. At the moment, there are 4 datasets available: Transcription Factor (TF)
	ChIP-seq from <a href="http://www.genome.ucsc.edu/ENCODE/dataMatrix/encodeChipMatrixHuman.html">ENCODE</a> in human (hg19),
    <a href="http://codex.stemcells.cam.ac.uk/about.php">CODEX</a> TF ChIP-seq for human (hg19) or mouse (mm10), and 
	<a href="http://www.modencode.org/">modENCODE</a> TF ChIP-seq for drosophila (r5).
</p>
<h3>
    <a name="_Loading_a_peak_file"></a>2.2 Loading a peak file
</h3>
<p>
	You can now upload a peak file (a list of genomic coordinates). Note that this is not mandatory, and you can jump to section
    <a href="#_Static_heatmap">2.4</a> if you simply want to browse the selected dataset. The application
	accepts a three column tab-delimited text file following a bed
    format: first column should be chromosome names (chr1, chr2, chrX, etc., or for drosophila 2L, 2R, 2LHet, etc.),
	the second should be genomic region start coordinate, and the third genomic region end coordinate.
	Any additional column
    will not be considered (peak name, score, strand, etc.). Please, make sure to untick the <strong>My peak file contains a header</strong>
	(does first line of the file contains column name?) option if your file does
    not contain a header. You can also fill the <strong>Name of your experiment</strong> field which will
    modify the label of your experiment.
</p>
<p class="text-muted">
    The maximum size you can upload is 10 Mb. If your bed file is larger than that, try keeping only the first three columns of it to reduce file size. If
	after removing the non-essential columns the file is still bigger than 10 Mb,
    <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a> (replace at with @).
</p>
<p>
    Once the file is uploaded, a subtle progress bar will appear on top of the page, and quick description of the on-going steps can be found on the top
    right of the page. It should take less than a minute to process about 30,000 genomic regions. The <strong>My peaks</strong>
	tab will display a table of the uploaded coordinates.
</p>
<p class="text-muted">
    The easiest way to remove a peak file without uploading a new one is to return to the main page and open a new HeatChIPseq window.
</p>
<p>
	First lines of your file should look like this:
	<table style="width:40%">
	  <tr>
		<th>chr</th>
		<th>start</th>
		<th>end</th>
	  </tr>
	  <tr>
		<td>chr1</td>
		<td>125423</td>
		<td>125891</td>
	  </tr>
	  <tr>
		<td>chr1</td>
		<td>8545032</td>
		<td>8546254</td>
	  </tr>
	  <tr>
		<td>chr4</td>
		<td>4523698</td>
		<td>4524785</td>
	  </tr>
	  <tr>
		<td>chr12</td>
		<td>854120</td>
		<td>854870</td>
	  </tr>
	  <tr>
		<td>chrX</td>
		<td>2458750</td>
		<td>2459872</td>
	  </tr>
	</table> 
</p>
<p>
	You can download 
	<a href="chipseq_human_hg19_GSM1890761_ERa_peaks_no_header.bed" download>this example file</a>.
	It is a ESR1 (ERalpha) ChIP-seq experiment from MCF7 cells, kindly provided by Dr. Str&ouml;mblad.
	Use a human dataset (for example ENCODE TFBS), and untick the "My peak file contains a header" option.
	It corresponds to the MCF7_ERa_E2_ChIP sample from 
    <a href= "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE73320")>GEO dataset GSE73320</a>
	supporting <a href="http://www.impactjournals.com/oncotarget/index.php?journal=oncotarget&page=article&op=view&path[]=6081">this study</a>.
	It is quite easy to rebuild. Goe to the MCF7_ERa_E2_ChIP <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM1890761" >sample page</a> and 
	download the GSM1890761_ERa_Input_peaks.bed.gz file. 
	Extract the archive, and you can now upload the GSM1890761_ERa_Input_peaks.bed to HeatChIPseq. 
</p>
<h3>
    <a name="_Looking_at_the_Correlation_table"></a>2.3 Looking at the Correlation table
</h3>
<p>
	If you have uploaded a peak file, the <strong>Correlation table</strong> tab should now display a 3 column table. The first column contains the name of
    <strong>experiments</strong> in the selected dataset, the second column contains the <strong>correlation</strong> coefficients between the uploaded
    coordinates and the corresponding experiment in the dataset. A third column contains <strong>scaled correlation</strong> (discussed in
    section <a href="#_Correlation_correction:">2.8</a>); this column is not visible by default. At the bottom of the page, you will find a <strong>Save as tab delimited .txt</strong> download
    button, allowing you to download a copy of the correlation table as a tab delimited text file. The file can be read by many spreadsheet programs
    including Microsoft Excel.
</p>
<p class="text-muted">
    You can copy an experiment name from the correlation table and paste it into the Search field of the Samples metadata tab to see the metadata available 
    for that experiment.
</p>
<h3>
    <a name="_Static_heatmap"></a>2.4 Static heatmap
</h3>
<p class="text-muted">
    The heatmap can take up to a minute to be displayed, and a bit longer if you are importing a new peak file or switching datasets.
</p>
<p>
    The <strong>static heatmap</strong> tab will display a clustered heatmap representation of the correlation matrix: each row and column represent a single
    experiment. The colour colour legend can be found below the heatmap: white represents a correlation coefficient equal to 0,
    red equals to 0.5, and black equals to 1. Several options are available on the side bar panel on the left, <strong>3 - Plot customization</strong>. In the
    order of appearance:
</p>
<ul>
    <li>
        The <strong>Highlight my experiment in the heatmap</strong> tick box. When checked, if you uploaded a peak file, the row and line corresponding to your
        experiment will be highlighted: a 0 correlation value will be displayed in yellow, a 0.5 value in bright green, and a 1 correlation value still in
        black. This can help you localise your experiment in crowded heatmaps.
    </li>
    <li>
        One or several <strong>subset</strong> options: You can restrict the experiments to be displayed using those fields. They may vary a bit between
		datasets. You can select experiments for one or several <strong>transcription factors</strong> (TFs), as well as select experiments done in one or several 
        <strong>cell types</strong>. Leaving the field empty will select all experiments.
    </li>
</ul>
<p class="text-muted">
    If less than three experiments from the dataset match your criteria, the heatmap will not be displayed. You can look for available experiments in the Samples metadata tab.
</p>
<p class="text-muted">
    The uploaded peak list is not affected by the filters and will always be displayed in the heatmap.
</p>
<ul>
    <li>
		The function of the <strong>Uploaded experiment correlation correction</strong> field and the <strong>Maximum expected correlation value for linear scaling correction</strong> slider is explained in section
        <a href="#_Correlation_correction:">2.8</a>.
    </li>
    <li>
		The <strong>Advanced clustering options</strong> button will reveal the <strong>Distance calculation</strong> and <strong>Clustering method</strong> options.
		The selected values are passed to the R <a href="https://stat.ethz.ch/R-manual/R-patched/library/stats/html/dist.html">dist()</a>
		and <a href="https://stat.ethz.ch/R-manual/R-patched/library/stats/html/hclust.html">hclust()</a> method parameter. A special case is the
        <strong>1 - Pearson's correlation coefficient</strong> distance method: instead of using the dist() function on correlation values, it uses 1 - correlations as a measure of
        distance between experiments (a correlation of 0.75 will give a distance of 0.25).
    </li>
    <li>
        The <strong>Sample name size</strong> slider will modify label size.
    </li>
    <li>
        The <strong>Sample name margin</strong> will modify the size of the margin of the plot.
    </li>
</ul>
<p class="text-muted">
    If you see an error message: <strong>Figure margins too large</strong>, try reducing the size of the margin as well as the size of the labels.
</p>
<p>
    Buttons <strong>Save as png</strong>, <strong>Save as pdf</strong> and <strong>Save as svg</strong> can be found below the heatmap to export the
    image in those formats. Button <strong>Export data as tab delimited .txt</strong> exports the heatmap data as a text file.
</p>
<h3>
    <a name="_Responsive_heatmap"></a>2.5 Responsive heatmap
</h3>
<p>
    The <strong>responsive heatmap </strong>tab displays an interactive plot provided through the <a href="https://plot.ly/">plot.ly</a> API. It can take 
    about a minute to be displayed on powerful computer running recent version of Firefox, Chrome, Safari, Internet explorer
    or Edge web browsers. It represents the same version of the <strong>Static heatmap</strong> presented in the previous section, without the side
    dendrograms (trees). Most of the options are the same as for <strong>Static heatmap</strong>, so please check section <a href="#_Static_heatmap">2.4</a>
    for more information. Additional options are available on the responsive heatmap itself:
</p>
<ul>
    <li>
    Mouse over information: when hovering the mouse over the heatmap, a text box will appear, giving you the name of the experiment in the x-axis, the
    name of the experiment in the y-axis, and their correlation value (z).
    </li>
</ul>
<p class="text-muted">
    If the Highlight my experiment in the heatmap option is enabled, the z value of the highlighted cells will be offset by 2: a correlation value of 0.75 will
    have a z-value of 2.75.
</p>
<ul>
    <li>
		One can zoom on the heatmap by dragging and dropping the mouse defining a rectangle. To zoom out, double-click on the heatmap or click the
		<strong>Reset axis</strong> or <strong>Autoscale</strong> buttons on the top left corner of the heatmap.
	</li>
    <li>
        By selecting the <strong>Pan</strong> button on the top left corner of the heatmap, dragging and dropping the mouse will allow to pan, which can be
        useful once you have zoomed in a specific part of the heatmap.
    </li>
    <li>
        The <strong>Download plot as a png</strong> button is non-functional (blame plot.ly). Please use the one under the <strong>Static heatmap</strong> or
        look at the next step.
    </li>
    <li>
        The <strong>Save and edit plot in the cloud</strong> will send the data to <a href="https://plot.ly/">plot.ly</a> where, after the loading, one can play with many tools offered by
        plot.ly, such as export a JSON version of the data, save the plot as a png, change the theme (and notably the colour scale), etc.
    </li>
</ul>
<h3>
    <a name="_Tree"></a>2.6 Tree
</h3>
<p>
    The <strong>Tree</strong> tab will display only the dendrogram (or Tree) from the experiment clustering. Options are mostly similar to the one for static
    heatmap, please refer to section <a href="#_Static_heatmap">2.4</a>. The <strong>Highlight my experiment in the heatmap</strong>
    option will not highlight your experiment in the dendrogram.
</p>
<p class="text-muted">
    If you see a red, unfriendly error message: <strong>Figure margins too large</strong>, try reducing the size of the margin as well as the name size.
</p>
<h3>
    <a name="_Samples_metadata"></a>2.7 Sample metadata
</h3>
<p>
	The sample metadata tab will display metadata information (experience name, cell type, url of the original data, etc.) for the <strong>selected dataset</strong>.
	The table is sortable and searchable, and can be downloaded as a tab-delimited txt file using the <strong>save</strong>
    button below the table.
</p>
<p>
    You can copy an experiment name from the correlation table and paste it into the Search field of the Samples metadata tab to learn all the metadata we have
    for that experiment.
</p>
<h3>
    <a name="_Correlation_correction:"></a>2.8 Correlation correction
</h3>
<p>
    Sometimes, the maximum correlation of a user peak file with any experiment in the dataset can be quite low. In some cases, when the user is confident
    that the top hits are relevant, this may be evidence of strong "batch effect" that could reflect an artefact of library preparation method, peak calling or
	ChIP efficiency. The low correlation values might bias the clustering due to <a href="https://en.wikipedia.org/wiki/Long_branch_attraction">Long Branch Attraction</a>.
	We provide an option to correct for his bias using the <strong>Linear scaling</strong> method on the <strong>Uploaded experiment correlation correction</strong> option.
	The correlation values will be linearly	up-scaled so the maximum correlation value will now be equal to the value of the
    <strong>Maximum expected correlation value for linear scaling correction </strong>slider (default 0.95). The resulting transformed correlation value can be
    obtained from the <strong>scaledCorrelation</strong> column of the <strong>Correlation table</strong>.
</p>
<p class="text-muted">
    The linear scaling of correlation value will not change the ordering of the values, only scales the value (i.e. the third most correlated experiment without scaling
	will still be third with scaling).
</p>
<h3>
	<a name="_About_the_datasets"></a>2.9 About the datasets
</h3>
<p><table style="width:100%">
 <tr>
  <th>Dataset</th>
  <th>Organism</th>
  <th>Number of experiments</th>
  <th>download date</th>
  <th>data from</th>
 </tr>
 <tr>
  <td>ENCODE</td>
  <td>human</td>
  <td>690</td>
  <td>2015-02-01</td>
  <td><a href="http://hgdownload.cse.ucsc.edu/goldenPath/hg19/encodeDCC/wgEncodeRegTfbsClustered/">ENCODE at UCSC</a></td>
 </tr>
 <tr>
  <td>CODEX</td>
  <td>human</td>
  <td>238</td>
  <td>2016-02-04</td>
  <td><a href="http://codex.stemcells.cam.ac.uk/about.php">CODEX</a></td>
 </tr>
 <tr>
  <td>CODEX</td>
  <td>mouse</td>
  <td>651</td>
  <td>2015-09-04</td>
  <td><a href="http://codex.stemcells.cam.ac.uk/about.php">CODEX</a></td>
 </tr>
 <tr>
  <td>modENCODE</td>
  <td>drosophila</td>
  <td>85</td>
  <td>2016-03-21</td>
  <td><a href="ftp://data.modencode.org/D.melanogaster/Transcriptional-Factor/ChIP-seq/computed-peaks_gff3/ ">modENCODE ftp</a></td>
 </tr>
</table></p>
<p>
    If you would like us to add a dataset, or to update an existing one, please <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a>
	(replace at with @). A well curated dataset can be implemented / updated within a working day.
</p>
<div>
    <h1>
        <a name="_FAQ"></a>3 FAQ
    </h1>
</div>
<h3>
    <a name="_How_to_cite"></a>3.1 How to cite?
</h3>
<p>
    Please cite this preprint: <a href="http://biorxiv.org/content/early/2016/04/18/049254">Heat*seq: an interactive web tool for high-throughput sequencing experiment comparison with public data</a>
</p>
<h3>
    <a name="_Where_can_I_find_the_source_code"></a>3.2 Where can I find the source code?
</h3>
<p>
    The source code is available on <a href="https://github.com/gdevailly/HeatStarSeq_gh">GitHub</a>.
</p>
<h3>
    <a name="_Can_you_add_this_dataset_on_HeatChIPseq"></a>3.3 Can you add this dataset on HeatChIPseq? One of your datasets does not seem up to date, can you update it?
</h3>
<p>
    Please <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a> (replace at with @). A well curated dataset can be
    implemented / updated within a working day.
</p>
<h3>
    <a name="_Could_you_implement_this_new_feature"></a>3.4 Could you implement this new feature? | I want to report a bug.
</h3>
<p>
    Please <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a> (replace at with @). We will be very pleased to consider implementing any
    feature that will improve the usability of this application.
</p>
<h3>
    <a name="_I_have_uploaded_a_peak_file"></a>3.5 I have uploaded a peak file. How do I remove it?
</h3>
<p>
    To remove a peak file without uploading a new one, the simplest method is to open a new HeatChIPseq session by going <strong>Back to the main page</strong>.
    Refreshing the HeatChIPseq page may result in a slightly erratic outcome.
	One can always replace it with any other peak file clicking the <strong>Browse</strong> button again.
</p>

<p>
	Uploaded files are stored in a temporary folder in our server. They are automaticaly deleted once the R session expires (i.e. when you close the window).
</p>
<div>
    <h1>
        <a name="_About"></a>4 About
    </h1>
</div>
<p>
    <strong>HeatChIPseq</strong>
    is a part of <strong>Heat*Seq</strong>, an attempt to make genome-wide comparison of high throughput sequencing experiments easier. It was developed by
	Guillaume Devailly, Anna Mantsoki and Anagha Joshi at the <a href="http://www.roslin.ed.ac.uk/">Roslin Institute</a>, and funded by the
	<a href="http://www.bbsrc.ac.uk/">Biotechnology and Biological Sciences Research Council</a>. It uses <a href="https://www.r-project.org/">R</a>
	<a href="https://www.rstudio.com/products/shiny/">shiny</a>, <a href="https://plot.ly/">plot.ly</a>, and various <a href="https://cran.r-project.org/">CRAN</a>
	and <a href="http://www.bioconductor.org/">Bioconductor</a> packages, and datasets from <a href="http://www.genome.ucsc.edu/ENCODE/dataMatrix/encodeChipMatrixHuman.html">ENCODE</a>, 
	<a href="http://codex.stemcells.cam.ac.uk/about.php">CODEX</a> and <a href="http://www.modencode.org/">modENCODE</a>. Sources are available on <a href="https://github.com/gdevailly/HeatStarSeq_gh">GitHub</a>.
</p>
</div>