<div id="instructions">
<div>
    <h1>
        <a name="_Contents"></a>1 Contents
    </h1>
</div>
<h4 style="padding-left: 1em;">
    <a href="#_Instructions">2 Instructions</a>
</h4>
<p style="padding-left: 3em;">
    <a href="#_Remarks_on_performance">2.1 Remarks on performance</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Selecting_a_dataset">2.2 Selecting a dataset</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Loading_a_CAGE_result_bed_file">2.3 Loading a CAGE result bed file</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Looking_at_the_Correlation_table">2.4 Looking at the Correlation table</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Static_heatmap">2.5 Static heatmap</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Responsive_heatmap">2.6 Responsive heatmap</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Tree">2.7 Tree</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Pairwise_plot">2.8 Pairwise plot</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Samples_metadata">2.9 Sample metadata</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Correlation_correction">2.10 Correlation correction</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_About_the_datasets">2.11 About the datasets</a>
</p>
<h4 style="padding-left: 1em;">
    <a href="#_FAQ">3 FAQ</a>
</h4>
<p style="padding-left: 3em;">
    <a href="#_How_to_cite">3.1 How to cite?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Where_can_I_find_the_source_code">3.2 Where can I find the source code?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Can_you_add_this_dataset_on_HeatChIPseq">3.3 Can you add this dataset on HeatCAGEseq? One of your datasets does not seem up to date, can you update it?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_Could_you_implement_this_new_feature">3.4 Could you implement this new feature? | I want to report a bug.</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_I_have_uploaded_a_peak_file">3.5 I have uploaded a CAGE result bed file. How do I remove it?</a>
</p>
<p style="padding-left: 3em;">
    <a href="#_What_happens_to_the_files_I_upload">3.6 What happens to the files I upload?</a>
</p>
<h4 style="padding-left: 1em;">
    <a href="#_About">4 About</a>
</h4>
<img src="screenshot.png" alt="HeatCAGEseq screenshot" width="800"> 
<div>
    <h1>
        <a name="_Instructions"></a>2 Instructions
    </h1>
</div>
<p>
    To access the application, click on the <strong>Use Application</strong> button on the <strong>top bar</strong>.
	The step-by-step instructions for using the application are below.
</p>
<h3>
    <a name="_Remarks_on_performance"></a>2.1 Remarks on performance
</h3>
<p>
	HeatRNAseq runs some quite intensive tasks on both server and client sides. It thus needs a reasonably recent browser
	on a reasonably fast computer, tablet of phone. Here are some general suggestions concerning performance:
</p>
<ul>
    <li>
		Most of the tasks are queued: changing parameters during loading does not stop current tasks
		but puts them on hold until all previous tasks are completed. The direct consequence is that some buttons
		might look non-responsive because the application is running a task in the background 
		(loading bar on top of the webpage). We suggest <b>to wait until current tasks are fully executed before changing parameters</b>.
	</li>
	<li>
		One particular intensive task is the <b>Interactive heatmap</b>, we recommend using the <b>static heatmap</b> first while
		changing options and filters before switching to the interactive heatmap.
	</li>
	<li>
		If the application shows a grey filter, it means the connection with the server was lost, sorry. :(
		Nothing can be done but re-launching the application. Refreshing the page usually keeps options as they were,
		while opening a new web page will start the application with default values.
	</li>
</ul>
<h3>
    <a name="_Selecting_a_dataset"></a>2.2 Selecting a dataset
</h3>
<p>
    The first step is to <strong>select a dataset</strong> to work with. At the moment, there are 2 datasets available: CAGE 
	experiments from <a href="http://fantom.gsc.riken.jp/">FANTOM5</a> in human (hg19) and mouse (mm9).
</p>
<h3>
    <a name="_Loading_a_CAGE_result_bed_file"></a>2.3 Loading a CAGE result bed file
</h3>
<p>
	You can now upload a CAGE result bed file. Note that this is not mandatory, and you can jump to section
    <a href="#_Static_heatmap">2.5</a> if you simply want to browse the selected dataset. The application accepts a six column tab-delimited text file following a bed
    format, and requires at least 6 columns: first column should be chromosome name (chr1, chr2, chrX, etc.),
	the second genomic region start coordinate, the third genomic region end coordinate,
	the fourth region name (not used, but required), the fifth a score CAGE expression value, such as RPM values or tag count, and the sixth 
	strand information. Any additional column 
    will not be considered. Please, make sure to untick the <strong>My bed file contains a header</strong> (does first line of the file contains column name?)
	option if your file does
    not contain a header. You can also fill the <strong>Name of your experiment</strong> field which will
    modify the label of your experiment.
</p>
<p class="text-muted">
    HeatRNAseq computes Pearson's correlation coefficient between experiments, after scaling them using log10(expression value + 1). Don't upload log-scaled values
	as the application will do it itself.
</p>
<p class="text-muted">
    The maximum size you can upload is 10 Mb. If your bed file is larger than that, try keeping only the first six columns of it to reduce file size. If
	after removing the non-essential columns the file is still bigger than 10 Mb,
    <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a> (replace at with @).
</p>
<p>
    Once the file is uploaded, a subtle progress bar will appear on top of the page, and quick description of the on-going steps can be found on the top
    right of the page. It should take a minute to process 100.000 regions. The <strong>My bed file</strong> tab will display a view of the uploaded file.
</p>
<p class="text-muted">
    The easiest way to remove a bed file without uploading a new one is to retrun to the main page and open a new HeatCAGEseq window.
</p>
<p>
	First lines of your file should look like this:
	<table style="width:100%">
	  <tr>
		<th>chr</th>
		<th>start</th>
		<th>end</th>
		<th>name</th>
		<th>rpm</th>
		<th>strand</th>
	  </tr>
	  <tr>
		<td>chr2</td>
		<td>25486325</td>
		<td>25486487</td>
		<td>CAGEpeak_1</td>
		<td>458.12</td>
		<td>+</td>
	  </tr>
	  <tr>
		<td>chr6</td>
		<td>5896321</td>
		<td>5896380</td>
		<td>CAGEpeak_2</td>
		<td>25.03</td>
		<td>+</td>
	  </tr>
	  <tr>
		<td>chr6</td>
		<td>223541</td>
		<td>223602</td>
		<td>CAGEpeak_3</td>
		<td>1.23</td>
		<td>-</td>
	  </tr>
	  <tr>
		<td>chr17</td>
		<td>5012035</td>
		<td>5012100</td>
		<td>CAGEpeak_4</td>
		<td>45.3</td>
		<td>+</td>
	  </tr>
	  <tr>
		<td>chr21</td>
		<td>960032</td>
		<td>960098</td>
		<td>CAGEpeak_5</td>
		<td>8.70</td>
		<td>-</td>
	  </tr>
	</table> 
</p>
<p>
	You can download 
	<a href="Expression.mm9.HCC.49096peaks_column1.bed" download>this example file</a>.
	It is a CAGE experiment from mouse liver cells, kindly provided by Dr. Carninci and colleagues.
	Use the mouse dataset, and untick the "My peak file contains a header" option.
	It corresponds to the WT1 sample from 
    <a href= "http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60982")>GEO dataset GSE60982</a>
	supporting <a href="http://genome.cshlp.org/content/early/2015/10/27/gr.191031.115?top=1">this study</a>.
	It is quite easy to rebuild. Goe to the <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE60982">GEO page</a> and 
	download the GSE60982_Expression.mm9.HCC.49096peaks.txt.gz file. 
	Extract the archive, and replace the fifth column by values contained in the thirteenth column, then delete every column but the six first. 
	The file is now ready to be upload to HeatCageSeq.
</p>
<h3>
    <a name="_Looking_at_the_Correlation_table"></a>2.4 Looking at the Correlation table
</h3>
<p>
	If you uploaded a bed file, the <strong>Correlation table</strong> tab should now display a 3 column table. The first column contains the name of
    <strong>experiments</strong> in the selected dataset, the second column contains the <strong>correlation</strong> coefficient between the uploaded
    coordinates and the corresponding experiment in the dataset. A third column contains <strong>scaled correlation</strong> values (discussed in
    section <a href="#_Correlation_correction:">2.10</a>); this column is not visible by default. At the bottom of the page, you will find a <strong>Save as tab delimited .txt</strong> download
    button, allowing you to download a copy of the correlation table as a tab delimited text file. Those files can be read by many spreadsheet programs
    including Microsoft Excel.
</p>
<p class="text-muted">
    You can copy an experiment name from the correlation table and paste it into the Search field of the Samples metadata tab to see the metadata available 
    for that experiment.
</p>
<h3>
    <a name="_Static_heatmap"></a>2.5 Static heatmap
</h3>
<p class="text-muted">
    The heatmap can take up to a minute to be displayed, and a bit longer if you are importing a new peak file or switching datasets.
</p>
<p>
    The <strong>static heatmap</strong> tab will display a clustered heatmap representation of the correlation matrix: each row and column represent a single
    experiment. The colour legend can be found below the heatmap: white represents a correlation coefficient equal to 0,
    red equals to 0.5, and black equals to 1. Several options are available on the side bar panel on the left, <strong>3 - Plot customixation</strong>. In the
    order of appearance:
</p>
<ul>
    <li>
        The <strong>Highlight my experiment in the heatmap</strong> tick box. When checked, if you uploaded a bed file, the row and line corresponding to your
        experiment will be highlighted: a 0 correlation value will be displayed in yellow, a 0.5 value in bright green, and a 1 correlation value still in
        black. This can help you localise your experiment in crowded heatmaps.
    </li>
    <li>
        One or several <strong>subset</strong> options: You can restrict the experiments to be displayed using those fields. They may vary a bit between
		datasets. You can usualy select experiments done in one or several  
        <strong>cell types</strong>. Leaving the field empty will select all experiments.
    </li>
</ul>
<p class="text-muted">
    If less than three experiments from the dataset match your criteria, the heatmap will not be displayed. You can look for available experiments in the Samples metadata tab.
</p>
<p class="text-muted">
    The uploaded experiment is not affected by the filters and will always be displayed in the heatmap.
</p>
<ul>
    <li>
		The function of the <strong>Uploaded experiment correlation correction</strong> field and the <strong>Maximum expected correlation value for linear scaling correction</strong> slider is detailed in section
        <a href="#_Correlation_correction:">2.10</a>.
    </li>
    <li>
		The <strong>Advanced clustering options</strong> button will reveal the <strong>Distance calculation</strong> and <strong>Clustering method</strong> options.
		The selected values are passed to the R <a href="https://stat.ethz.ch/R-manual/R-patched/library/stats/html/dist.html">dist()</a>
		and <a href="https://stat.ethz.ch/R-manual/R-patched/library/stats/html/hclust.html">hclust()</a> method parameter. A special case is the
        <strong>1 - Pearson's correlation coefficient</strong> distance method: instead of using the dist() function on correlation values, it uses 1 - correlations as a measure of
        distance between experiments (a correlation of 0.75 will give a distance of 0.25).
    </li>
	<li>
       Label and margin size can be decided manually: The <strong>Sample name size</strong> slider will modify label size.
	   The <strong>Sample name margin</strong> will modify the size of the margin of the plot.
	   Two scrolling menus will allow to show and hide row and/or column dendrograms and sample labels.
    </li>
    <li>
		The <strong>Customise colour</strong> button will revealed items allowing to change the colour key used for the heatmap.
		Four threshold values and associated colours can be specified (by default, 0.25 is blue, 0.5 is white, 0.75 is red, and 1 is black).
		Values need to be strictly increasing. To apply changes, click the <strong>Apply colour changes</strong> button.
    </li>
</ul>
<p class="text-muted">
    If you see an error message: <strong>Figure margins too large</strong>, try reducing the size of the margin as well as the size of the labels.
</p>
<p>
    Buttons <strong>Save as png</strong>, <strong>Save as pdf</strong> and <strong>Save as svg</strong> can be found below the heatmap to export the
    image in those formats. Button <strong>Export data as tab delimited .txt</strong> exports the heatmap data as a text file.  To download the colour key,
	right click on it and select <strong>Save Image As...</strong>
</p>
<h3>
    <a name="_Responsive_heatmap"></a>2.6 Responsive heatmap
</h3>
<p>
    The <strong>responsive heatmap </strong>tab displays an interactive plot provided through the <a href="https://plot.ly/">plot.ly</a> API. It can take
    about a minute to be displayed on powerful computer running recent version of Firefox, Chrome, Safari, Internet explorer or
    Edge. It represents the same version of the <strong>Static heatmap</strong> presented in the previous section, without the side
    dendrograms (trees). Most of the options are the same as for <strong>Static heatmap</strong>, so please check section <a href="#_Static_heatmap">2.5</a>
    for more information. Additional options are available on the responsive heatmap itself:
</p>
<ul>
    <li>
    Mouse over information: when hovering the mouse over the heatmap, a text box will appear, giving you the name of the experiment in the x-axis, the
    name of the experiment in the y-axis, and their correlation value (z).
    </li>
</ul>
<p class="text-muted">
    If the Highlight my experiment in the heatmap option is enabled, the z value of the highlighted cells will be offset by 2: a correlation value of 0.75 will
    have a z-value of 2.75.
</p>
<ul>
    <li>
	One can zoom on the heatmap by dragging and dropping the mouse defining a rectangle. To zoom out, double-click on the heatmap or click the
    <strong>Reset axis</strong> or <strong>Autoscale</strong> buttons on the top left corner of the heatmap.
    </li>
    <li>
        By selecting the <strong>Pan</strong> button on the top left corner of the heatmap, dragging and dropping the mouse will allow to pan, which can be
        useful once you have zoomed in a specific part of the heatmap.
    </li>
    <li>
        The <strong>Download plot as a png</strong> button is non-functional (blame plot.ly). Please use the one under the <strong>Static heatmap</strong> or
        look at the next step.
    </li>
    <li>
        The <strong>Save and edit plot in the cloud</strong> will send the data to <a href="https://plot.ly/">plot.ly</a> where, after the loading, one can play with many tools offered by
        plot.ly, such as export a JSON version of the data, save the plot as a png, change the theme (and notably the colour scale), etc.
    </li>
</ul>
<h3>
    <a name="_Tree"></a>2.7 Tree
</h3>
<p>
    The <strong>Tree</strong> tab will display only the dendrogram (or Tree) from the experiment clustering. Options are mostly similar to the one for static
    heatmap, please refer to section <a href="#_Static_heatmap">2.5</a>. The <strong>Highlight my experiment in the heatmap</strong>
    option will not highlight your experiment in the dendrogram.
</p>
<p class="text-muted">
    If you see a red, unfriendly error message: <strong>Figure margins too large</strong>, try reducing the size of the margin as well as the size of the name size.
</p>
<h3>
    <a name="_Pairwise_plot"></a>2.8 Pairwise plot
</h3>
<p>
	The <strong>Pairwise plot</strong> tab will display a scatter plot of two selected experiments where each dot represents a CAGE peak.
	Under the <strong>Plot customisation</strong> section on the left of the page, you can:
<ul>
	<li> choose experiment 1 and experiment 2.</li>
	<li> choose the plot type, between XY (where x-axis will be expression level in experiment 1, and y-axis expression level in experiment 2) and MA 
	(where x-axis will be mean expression level, and y-axis expression level in experiment 1 minus expression level in experiment 2).</li>
	<li> choose a different scaling to apply to the expression data. By default, if <br>e</br> is the expression value, HeatCAGEseq will apply a logscale log10(e + 1) before computing the Pearson
	correlation coefficient. You can select different scaling here to see the impact on the Pearson correlation coefficient.
	<li> add or remove a linear regression line (blue).
	<li> add or remove a guide line (red), which will be y=x in XY plot, and y=0 in MA plot.</li>
</ul>
	Below the scatterplot should be the Pearson and Spearman correlation coefficients between the two experiments, computed after scaling.
</p>
<p class="text-muted">
    Spearman correlation coefficient is independent of scaling, as scaling methods proposed here do not change the order of the data.
	All log transformations of the data have the same Pearson correlation coefficient as log in one base is a linear transformation of log in a different base.
</p>
<p>
	Further bellow are buttons to download the scatter plot as png, pdf or svg (recommended), as well as to export a tabulated text file containing the data used.
</p>
<h3>
    <a name="_Sample_metadata"></a>2.9 Sample metadata
</h3>
<p>
	The sample metadata tab will display metadata information (experiment name, cell type, url of the original data, etc.) for the <strong>selected dataset</strong>.
	The table is sortable and searchable, and can be downloaded as a tab-delimited txt file using the <strong>save</strong> button below the table.
</p>
<p>
    You can copy an experiment name from the correlation table and paste it into the Search field of the Samples metadata tab to show all the metadata we have
    for that experiment.
</p>
<h3>
    <a name="_Correlation_correction"></a>2.10 Correlation correction
</h3>
<p>
    Sometimes, the maximum correlation of a user peak file with any experiment in the dataset can be quite low. In some cases, when the user is confident
    that the top hits are relevant, this may be evidence of strong "batch effect" that could reflect an artefact of library preparation method or diferences in bioinformatic pipeline.
	The low correlation values might bias the clustering due to <a href="https://en.wikipedia.org/wiki/Long_branch_attraction">Long Branch Attraction</a>.
	We provide an option to correct for this bias using the <strong>Linear scaling</strong> method on the <strong>Uploaded experiment correlation correction</strong> option.
	The correlation values will be linearly	up-scaled so the maximum correlation value will now be equal to the value of the
    <strong>Maximum expected correlation value for linear scaling correction</strong> slider (default 0.95). The resulting transformed correlation value can be
    obtained from the <strong>scaledCorrelation</strong> column of the <strong>Correlation table</strong>.
</p>
<p class="text-muted">
    The linear scaling of correlation value does not change the ordering of the values, it only scales the value (i.e. the third most correlated experiment without scaling
	will still be third with scaling).
</p>
<h3>
    <a name="_About_the_datasets"></a>2.11 About the datasets
</h3>
<p>
<table style="width:100%">
 <tr>
  <th>Dataset</th>
  <th>Organism</th>
  <th>Number of experiments</th>
  <th>download date</th>
  <th>data from</th>
 </tr>
 <tr>
  <td>Fantom5</td>
  <td>human</td>
  <td>1058</td>
  <td>2015-11-04</td>
  <td><a href="http://fantom.gsc.riken.jp/5/datafiles/latest/extra/CAGE_peaks/">Fantom5</a></td>
 </tr>
 <tr>
  <td>Fantom5</td>
  <td>mouse</td>
  <td>490</td>
  <td>2015-11-04</td>
  <td><a href="http://fantom.gsc.riken.jp/5/datafiles/latest/extra/CAGE_peaks/ ">Fantom5</a></td>
 </tr>
</table></p>
<p>
    If you would like us to add a dataset, or to update an existing one, please <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a>
	(replace at with @). A well curated dataset can be implemented / updated within a working day.

</p>
<div>
    <h1>
        <a name="_FAQ"></a>3 FAQ
    </h1>
</div>
<h3>
    <a name="_How_to_cite"></a>3.1 How to cite?
</h3>
<p>
    Please cite this paper: <a href="https://academic.oup.com/bioinformatics/article/32/21/3354/2415180/Heat-seq-an-interactive-web-tool-for-high">Heat*seq: an interactive web tool for high-throughput sequencing experiment comparison with public data</a>
</p>
<h3>
    <a name="_Where_can_I_find_the_source_code"></a>3.2 Where can I find the source code?
</h3>
<p>
    The source code is available on <a href="https://github.com/gdevailly/HeatStarSeq_gh">GitHub</a>.
</p>
<h3>
    <a name="_Can_you_add_this_dataset_on_HeatChIPseq"></a>3.3 Can you add this dataset on HeatCAGEseq? One of your datasets does not seem up to date, can you update it?
</h3>
<p>
    Please <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a> (replace at with @). A well curated dataset can be
    implemented / updated within a working day.
</p>
<h3>
    <a name="_Could_you_implement_this_new_feature"></a>3.4 Could you implement this new feature? | I want to report a bug.
</h3>
<p>
    Please <a href="mailto:guillaume.devailly%20at%20roslin.ed.ac.uk">contact us</a> (replace at with @). We will be very pleased to consider implementing any
    feature that will improve the usability of this application.
</p>
<h3>
    <a name="_I_have_uploaded_a_peak_file"></a>3.5 I have uploaded a bed file. How do I remove it?
</h3>
<p>
    To remove a bed file without uploading a new one, the simplest method is to open a new HeatCAGEseq session by going <strong>Back to the main page</strong>.
    Refreshing the HeatCAGEseq page may result in slightly erratic outcome.
	One can always replace it with any other bed file clicking the <strong>Browse</strong> button again.
</p>
<h3>
    <a name="_What_happens_to_the_files_I_upload"></a> 3.6 What happens to the files I upload?
</h3>
<p>
	Uploaded files are stored in a temporary folder in our server. They are automaticaly deleted once the R session expires (i.e. when you close the window).
</p>
<div>
    <h1>
        <a name="_About"></a>4 About
    </h1>
</div>
<p>
    <strong>HeatCAGEseq</strong>
    is a part of <strong>Heat*Seq</strong>, an attempt to make genome-wide comparison of high throughput sequencing experiments easier. It was developed by
	Guillaume Devailly, Anna Mantsoki and Anagha Joshi at the <a href="http://www.roslin.ed.ac.uk/">Roslin Institute</a>, and funded by the
	<a href="http://www.bbsrc.ac.uk/">Biotechnology and Biological Sciences Research Council</a>. It uses <a href="https://www.r-project.org/">R</a>
	<a href="https://www.rstudio.com/products/shiny/">shiny</a>, <a href="https://plot.ly/">plot.ly</a>, and various <a href="https://cran.r-project.org/">CRAN</a>
	and <a href="http://www.bioconductor.org/">Bioconductor</a> packages, and datasets from <a href="http://fantom.gsc.riken.jp/5/">FANTOM5</a>.
	Sources are available on <a href="https://github.com/gdevailly/HeatStarSeq_gh">GitHub</a>.
</p>
</div>